File: src/converter/plugins/convertParenthesizedType.ts
import ts from "typescript";
import {createSimplePlugin} from "../plugin.js";
import {CheckCoverageService, checkCoverageServiceKey} from "./CheckCoveragePlugin.js";

export const convertParenthesizedType = createSimplePlugin((node, context, render) => {
    if (!ts.isParenthesizedTypeNode(node)) return null

    const checkCoverageService = context.lookupService<CheckCoverageService>(checkCoverageServiceKey)
    checkCoverageService?.cover(node)

    return `(${render(node.type)})`
})

File: karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/extension/plugins/convertParenthesizedType.kt
package io.github.sgrishchenko.karakum.extension.plugins

import io.github.sgrishchenko.karakum.extension.createPlugin
import typescript.isParenthesizedTypeNode

val convertParenthesizedType = createPlugin plugin@{ node, context, render ->
    if (!isParenthesizedTypeNode(node)) return@plugin null

    val checkCoverageService = context.lookupService(checkCoverageServiceKey)
    checkCoverageService?.cover(node)

    "(${render(node.type)})"
}
